dusty/command_file.py

```diff
@@ -74,49 +79,36 @@ def dusty_command_file_name(app_or_lib_name, script_name=None, test_name=None):
+            commands = ["cd {}".format(container_code_path(spec))] + script_spec['command']
             commands[-1] = '{} $@'.format(commands[-1])
-            _write_commands_to_file(commands, _get_command_file_path(app_name, assembled_specs, script_name=script_spec['name']))
-
-def remove_up_command_files(assembled_specs):
-    for app_name in assembled_specs['apps'].keys():
-        os.remove(_get_command_file_path(app_name, assembled_specs))
-        script_specs = assembled_specs['apps'][app_name]['scripts']
-        for script_spec in script_specs:
-            os.remove(_get_command_file_path(app_name, assembled_specs, script_name=script_spec['name']))
+            command_file_name = dusty_command_file_name(app_name, script_name=script_spec['name'])
+            local_path = '{}/{}/{}'.format(constants.COMMAND_FILES_DIR, app_name, command_file_name)
+            _write_commands_to_file(commands, local_path)
+    sync_local_path_to_vm(constants.COMMAND_FILES_DIR, constants.VM_COMMAND_FILES_DIR)
 
 def make_test_command_files(expanded_specs):
     for app_or_lib_spec in expanded_specs.get_apps_and_libs():
         test_spec = app_or_lib_spec['test']
         if test_spec and test_spec['once']:
             commands = _get_test_image_setup_commands(app_or_lib_spec.name, expanded_specs, test_spec)
-            _write_commands_to_file(commands, _get_command_file_path(app_or_lib_spec.name, expanded_specs))
+            command_file_name = dusty_command_file_name(app_or_lib_spec.name)
+            local_path = '{}/{}/test/{}'.format(constants.COMMAND_FILES_DIR, app_or_lib_spec.name, command_file_name)
+            _write_commands_to_file(commands, local_path)
             suite_specs = test_spec['suites']
             for suite_spec in suite_specs:
                 commands = ["cd {}".format(container_code_path(app_or_lib_spec))] + suite_spec['command']
                 commands[-1] = '{} $@'.format(commands[-1])
-                _write_commands_to_file(commands, _get_command_file_path(app_or_lib_spec.name, expanded_specs, test_name=suite_spec['name']))
+                command_file_name = dusty_command_file_name(app_or_lib_spec.name, test_name=suite_spec['name'])
+                local_path = '{}/{}/test/{}'.format(constants.COMMAND_FILES_DIR, app_or_lib_spec.name, command_file_name)
+                _write_commands_to_file(commands, local_path)
 
-def remove_test_command_files(expanded_specs):
-    for app_or_lib_spec in expanded_specs.get_apps_and_libs():
-        test_spec = app_or_lib_spec['test']
-        if test_spec and test_spec['once']:
-            os.remove(_get_command_file_path(app_or_lib_spec.name, expanded_specs))
-            suite_specs = test_spec['suites']
-            for suite_spec in suite_specs:
-                os.remove(_get_command_file_path(app_or_lib_spec.name, expanded_specs, test_name=suite_spec['name']))
+    sync_local_path_to_vm(constants.COMMAND_FILES_DIR, constants.VM_COMMAND_FILES_DIR)
```

Comment:
These functions are both pretty hairy, can they be broken down into smaller components?